hotfix and new feature

Temmie-Flakes opened the pull request:
This changes the jquery that find controls if you are logged in due to the UI change that breaks it.
All javascript is also now reformatted using tabs for indents.
New checkbox added in the config that makes the slideshow load in the highest resolution. This uses much more data and may cause the page to load in chunks so it is off by default (much better for things like comic reading).

Temmie-Flakes committed:
Bugfix login detection

The UI changed which caused code to reference an ID that no longer exists

Temmie-Flakes committed:
Merge branch 'umami-webext:master' into master

Temmie-Flakes committed:
Feature: High Resolution Images

Added a button in the config that enables loading the original post quality.
It is off by default to help with lag

Temmie-Flakes committed:
Reformat/beautify code using tabs

Using tabs just makes things more consistent for different code editors.

umami-webext commented:
Not generally opposed to the large images feature

But this can be achieved by changing the setting [in your preferences on e621 itself](https://e621.net/users/328443/edit#user_default_image_size). 

`Settings > Basic > Posts > Default image width`

Do you think making it an extra mode in umami is worth it? People wanting just their slideshow to be highres, but not general page visits?

If so then sure.

Temmie-Flakes commented:
Yea I only realized after I added it, that changing your preferences will make the slideshow change as it copies everything from the html. Problem is, people not logged in cannot change the default width, and I like to change it on the fly without having to go to a separate tab to change the resolution.
like if the compressed image is bad then I would either have to go into the page or go out of the tab to change my settings.

That being said I forgot to force it to be the compressed quality if you happen to have your default width at original
there needs to be an `else` that uses `data-large-url` rather then `data-file-url`

umami-webext commented:
Could it have 3 settings?

High rez
Low rez
Leave it as is

Temmie-Flakes commented:
Yea, like a dropdown selection of the 3 would be good. I will try to make something once I figure out how to make a dropdown.

Temmie-Flakes committed:
Feature and Bugfix of Image Resolutions

Changed the high res checkbox to be a dropdown menu
Bugfix of the default language not being updated in the dropdown menu
Bugfix of the high resolution skipping videos while on slideshow and autoplay.

Temmie-Flakes commented:
I think I finally got it. I had to do a a lot of weird stuff because videos use a different method of selecting resolution than images. but now its a dropdown of the three options. And it triggers a refresh of the post when you change it.

Temmie-Flakes committed:
Bugfix with low res causing cache to be deleted

Still ironing out this thing for a bit. Yea a bit unorthodox cuz of the jank way were using the e6API. Should be able to get this done soon

umami-webext commented:
Looks ok to me, I'll merge it into staging, and can have some people test it
